packages/amplify-graphql-auth-transformer/src/utils/index.ts

```diff
@@ -56,6 +65,30 @@ export const ensureAuthRuleDefaults = (rules: AuthRule[]) => {
   }
 };
 
+export const getAuthDirectiveRules = (authDir: DirectiveWrapper): AuthRule[] => {
+  const rules = authDir.getArguments<{ rules: Array<AuthRule> }>({ rules: [] }).rules;
+
+  for (const rule of rules) {
+    const operations: (ModelOperation | 'read')[] = rule.operations ?? [];
+
+    if (operations.includes('read') && (operations.includes('list') || operations.includes('get'))) {
+      const offendingOperation = operations.includes('list') ? 'list' : 'get';
+      throw new InvalidDirectiveError(
+        `'${offendingOperation}' operation is specified in addition to 'read'. Either remove 'read' to limit access only to '${offendingOperation}' or only keep 'read' to grant both 'get' and 'list' access.`,
+      );
+    }
+
+    if (operations.includes('read')) {
```

Comment:
Can you store the index in a variable. Then you no longer need the `includes()` call.

Reply:
updated